runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientProduceFactory.java

```diff
@@ -1723,6 +1725,16 @@ private void doEncodeRecordInit(
                                              .valueLength(valueLength)
                                              .build();
 
+                final MutableDirectBuffer encodeBuffer = writeBuffer;
+                this.recordHeaderSize = recordHeader.sizeof();
+                encodeBuffer.putBytes(0, encodeSlotBuffer, encodeSlotLimit, this.recordHeaderSize);
+
+                final byte[] encodeByteBuf = encodeBuffer.byteArray();
+                crc32c.reset();
+                crc32c.update(encodeByteBuf, 0, this.recordHeaderSize);
+
+                recordHeaderChecksum = crc32c.getValue();
+
```

Comment:
Can this be solved by adjusting the `crcDataLimit` to include the record header in `encodeCrc()` instead?

How would this work for record batches with multiple records?